openqaoa/workflows/optimizer.py

```diff
@@ -430,42 +458,149 @@ def compile(self, problem: QUBO = None, verbose: bool = True):
+            
+            # Ensure there are enough steps in the schedule
+            assert np.abs(n_qubits - n_cutoff - counter) <= sum(self.rqaoa_parameters.steps),\
+                f"""Schedule is incomplete, add 
+                    {np.abs(n_qubits - n_cutoff - counter) - sum(self.rqaoa_parameters.steps)} 
+                    more eliminations"""
+        
+        elif not self.rqaoa_parameters.rqaoa_type.lower() == "adaptive": 
+            
+            raise f'rqaoa_type {self.rqaoa_parameters.rqaoa_type} is not supported. Please selet either "adaptive" or "custom'
+        
+        #? verbose -> how we do?
+
+        return 
+
+    def optimize(self, verbose=False):
         """
         Performs optimization using RQAOA with the `custom` method or the `adaptive` method.
         """
-        if self.rqaoa_parameters.rqaoa_type == 'adaptive':
-            res = adaptive_rqaoa(
-                hamiltonian=self.qaoa.cost_hamil,
-                mixer=self.rqaoa_mixer,
-                p=self.qaoa.circuit_properties.p,
-                n_max=self.rqaoa_parameters.n_max,
-                n_cutoff=self.rqaoa_parameters.n_cutoff,
-                device=self.qaoa.device,
-                params_type=self.qaoa.circuit_properties.param_type,
-                init_type=self.qaoa.circuit_properties.init_type,
-                optimizer_dict=self.qaoa.classical_optimizer.asdict(),
-                backend_properties=self.qaoa.backend_properties.__dict__)
-        elif self.rqaoa_parameters.rqaoa_type == 'custom':
-            res = custom_rqaoa(
-                hamiltonian=self.qaoa.cost_hamil,
-                mixer=self.rqaoa_mixer,
-                p=self.qaoa.circuit_properties.p,
-                n_cutoff=self.rqaoa_parameters.n_cutoff,
-                steps=self.rqaoa_parameters.steps,
-                device=self.qaoa.device,
-                params_type=self.qaoa.circuit_properties.param_type,
-                init_type=self.qaoa.circuit_properties.init_type,
-                optimizer_dict=self.qaoa.classical_optimizer.asdict(),
-                backend_properties=self.qaoa.backend_properties.__dict__)
+
+        # reset all final variables
+        self.elimination_tracker = []
+        self.qaoa_steps = []
+        self.results = {}
+
+        # get variables
+        problem = self.problem  
+        n_cutoff = self.rqaoa_parameters.n_cutoff
+        n_qubits = problem.n
+        counter = self.rqaoa_parameters.counter 
+
+        # create a different max_terms function for each type 
+        if self.rqaoa_parameters.rqaoa_type.lower() == "adaptive":
+            f_max_terms = rqaoa.ada_max_terms  
         else:
-            raise f'rqaoa_type {self.rqaoa_parameters.rqaoa_type} is not supported. Please selet either "adaptive" or "custom'
+            f_max_terms = rqaoa.max_terms #? rqaoa.max_terms -> rqaoa.custom_max_terms ?
+
+        # If above cutoff, loop quantumly, else classically
+        while n_qubits > n_cutoff:
```

Comment:
I believe it could make sense to have a result object for this part of the algorithm and make sure that anything of interest in this while-loop is logged.

Probably the way to go is to have a logger here and then have something that formats the result to be presented to the user.

 Pinging @shahidee44, as he may have some good comments for this feature

Reply:
I think the RQAOAResults object has all the information to keep track of what has happened in each step (see the conversation below). I don't know if that is good enough, or if we should have a different object for that.

Reply:
Raul's RQAOAResults object is quite comprehensive. It might be worth reviewing what he has added. He's showed me some of the features that he has incooporated it and I believe its sufficient for our use-case. 
Unless you have anything else specific in mind, I think it looks good and is quite functional.

Reply:
Marking this as resolved